src/blog/better-class-features.md

```diff
@@ -0,0 +1,332 @@
+
+In most use cases, however, the semantic differences are insignificant. It would be nice to have the performance of the optimized assignments of properties when the pattern of the class field initialization is predictable enough - so a more optimal implementation was created after the proposal finalized.
+
+### Optimizing private class fields and computed public class fields
+
+To speed up initialization of private class fields and computed public class fields, the implementation introduced a new machinery to plug into the [inline cache (IC) system](https://mathiasbynens.be/notes/shapes-ics) when handling these operations, including:
+
+- A new bytecode `DefineKeyedOwnProperty`. This gets emitted when generating code for the `ClassLiteral::Property` AST nodes representing class field initializers.
+- A corresponding Turbofan IR opcode `JSDefineKeyedOwnProperty`, which can be compiled from the new bytecode.
+- A new `DefineKeyedOwnIC` which is used in the interpreter handler of the new bytecode as well as the code compiled from the new IR opcode. To simplify the implementation, the new IC reuses some of the code in `KeyedStoreIC` which was intended for ordinary property stores.
+
+Now when V8 encounters this class:
+
+```js
+class A {
+  #a = 0;
+}
+```
+
+It generates the following bytecode for the initializer `#a = 0`:
+
+```cpp
+// Load the private name symbol for `#a` into r1
+LdaImmutableCurrentContextSlot [2]
+Star0
+
+// Use the DefineKeyedOwnProperty bytecode to store 0 as the value of
+// the property keyed by the private name symbol `#a` in the instance,
+// that is, `#a = 0`.
+LdaZero
+DefineKeyedOwnProperty <this>, r0, [0]
+```
+
+When the initializer is executed for enough times, V8 allocates one [feedback vector slot](https://ponyfoo.com/articles/an-introduction-to-speculative-optimization-in-v8) for each field being initialized. The slot contains the key of the field being added (in the case of the private field, the private name symbol) and a pair of [hidden classes](https://v8.dev/blog/fast-properties) between which the instance has been transitioning as the result of field initialization. In subsequent initializations, the IC uses the feedback to see if the fields are initialized in the same order on instances with the same hidden classes. If the initialization matches the pattern that V8 has seen before (which is usually the case), V8 takes the fast path and performs the initialization with pre-generated code instead of calling into the runtime, thus speeding up the operation. If the initialization does not match a pattern that V8 has seen before, it falls back to a runtime call to deal with the slow cases.
```

Comment:
```suggestion
When the initializer is executed enough times, V8 allocates one [feedback vector slot](https://ponyfoo.com/articles/an-introduction-to-speculative-optimization-in-v8) for each field being initialized. The slot contains the key of the field being added (in the case of the private field, the private name symbol) and a pair of [hidden classes](https://v8.dev/docs/hidden-classes) between which the instance has been transitioning as the result of field initialization. In subsequent initializations, the IC uses the feedback to see if the fields are initialized in the same order on instances with the same hidden classes. If the initialization matches the pattern that V8 has seen before (which is usually the case), V8 takes the fast path and performs the initialization with pre-generated code instead of calling into the runtime, thus speeding up the operation. If the initialization does not match a pattern that V8 has seen before, it falls back to a runtime call to deal with the slow cases.
```

Camillo just wrote a new article. :)